src/ui/components/NetworkMonitor/index.tsx

```diff
@@ -111,23 +112,23 @@ export const NetworkMonitor = ({
                     dispatch(fetchRequestBody(row.id, row.point.point));
                   }
 
-                  setSelectedRequest(row);
+                  dispatch(showRequestDetails(row.id));
                 }}
                 seek={seek}
-                selectedRequest={selectedRequest}
+                selectedRequest={data.find(request => request.id === selectedRequestId)}
               />
             }
             endPanel={
-              selectedRequest ? (
+              selectedRequestId ? (
                 loadedRegions &&
-                getPointIsInLoadedRegion(loadedRegions, selectedRequest.point.point) ? (
+                getPointIsInLoadedRegion(loadedRegions, data.find(request => request.id === selectedRequestId)!.point.point) ? (
```

Comment:
:question: Looks like we've got `data.find()` multiple times.  Can we do that once earlier in the rendering logic and find the current request ahead of time?